Dark mode

ThomasdenH opened the pull request:
This is a PR for a minimal dark mode. Two things are open:

- The white in the red - white - green gradient is a little out of place.
- I would like to adapt the tests to also test dark mode. Unfortunately these test do not run for me, I think because `,` get turned into `.` and vice versa.

andreasgerstmayr commented on frontend/src/components/EChart.tsx:
```diff
@@ -30,6 +33,11 @@ export function EChart({ width, height, option }: EChartProps) {
       option.animation = false;
     }
 
+    option = {
```
```suggestion
    if (theme.palette.mode === "dark" && options.backgroundColor === undefined) {
      options.backgroundColor = "transparent";
    }
```

andreasgerstmayr commented on frontend/src/components/EChart.tsx:
```diff
@@ -30,6 +33,11 @@ export function EChart({ width, height, option }: EChartProps) {
       option.animation = false;
     }
 
+    option = {
+      ...option,
+      backgroundColor: '#0000',
+    }
```
```suggestion
    if (theme.palette.mode === "dark" && options.backgroundColor === undefined) {
      options.backgroundColor = "transparent";
    }
```

andreasgerstmayr commented on frontend/src/components/InvestmentsSelection.tsx:
```diff
@@ -103,14 +106,14 @@ export function InvestmentsSelection(props: InvestmentsSelectionProps) {
       }
       style={{ width: 600 }}
-      renderInput={(params) => <TextField {...params} label={label} />}
+      renderInput={(params) => <TextField {...params} label={label} className="intestment-selection"/>}
```
```suggestion
      renderInput={(params) => <TextField {...params} label={label} className="investment-selection"/>}
```

andreasgerstmayr commented:
LGTM, thanks for the changes!

> I would like to adapt the tests to also test dark mode. Unfortunately these test do not run for me, I think because `,` get turned into `.` and vice versa.

Can you try `make container-test`? This target runs the tests in a container, which gives reproducible results across all platforms and environments.

ThomasdenH commented:
I fixed your suggestions and the linting and formatting. I added tests for the dark mode, the light mode seems to be unaffected other than the random ids having been changed.

I decided against changing the returns gradient color, since pure white appears in the scheme already and grey didn't look right. It can be improved later on anyway.

Let me know if there is anything I should change!

andreasgerstmayr commented:
Looks good! Could you fix the formatting in `pages.test.ts` (see CI output)?
Then it's ready to be merged.

ThomasdenH commented:
Done!

andreasgerstmayr commented:
Could you also run `make build` and commit the `.js` file?
I know it's not ideal to store the compiled javascript in the git repo, but it makes installing this project much easier (via `pip install git+https://...`).

andreasgerstmayr commented:
Thanks!
